iceoryx_posh/source/runtime/ipc_interface_base.cpp

```diff
@@ -77,12 +78,47 @@ std::string IpcMessageErrorTypeToString(const IpcMessageErrorType msg) noexcept
     return convert::toString(static_cast<UnderlyingType>(msg));
 }
 
+iox::optional<RuntimeName_t> ipcChannelNameToInterfaceName(RuntimeName_t channelName)
+{
+    RuntimeName_t interfaceName = iceoryxResourcePrefix(roudi::DEFAULT_UNIQUE_ROUDI_ID);
+    if (interfaceName.size() + channelName.size() > RuntimeName_t::capacity())
+    {
+        return nullopt;
+    }
+    interfaceName.append(TruncateToCapacity, channelName);
+    return interfaceName;
+}
+
 template <typename IpcChannelType>
 IpcInterface<IpcChannelType>::IpcInterface(const RuntimeName_t& runtimeName,
                                            const uint64_t maxMessages,
                                            const uint64_t messageSize) noexcept
-    : m_runtimeName(runtimeName)
 {
+    if (runtimeName.empty())
+    {
+        IOX_PANIC("Then runtime name must not be empty");
+    }
+    for (const auto s : platform::IOX_PATH_SEPARATORS)
+    {
+        const char separator[2]{s};
+        if (runtimeName.find(separator).has_value())
+        {
+            IOX_LOG(FATAL, "The runtime name '" << runtimeName << "' contains path separators");
+            IOX_PANIC("Invalid characters for runtime name");
+        }
+    }
+
+    m_interfaceName =
+        ipcChannelNameToInterfaceName(runtimeName)
+            .or_else([&runtimeName] {
+                IOX_LOG(FATAL,
+                        "The runtime with the name '"
+                            << runtimeName.size()
+                            << "' would exceed the maximum allowed size when used with the 'iox1_#_' prefix!");
```

Comment:
Is there a function that returns just the prefix. This seems to be a bit error-prone. What if we change the prefix than this will be out of sync.

Reply:
Well, the function is a few lines further above `ipcChannelNameToInterfaceName` ;)

Reply:
No, I mean just the prefix. This function returns the whole name with the prefix.

Reply:
That's what `iceoryxResourcePrefix` does :)